types/i18next/index.d.ts

```diff
@@ -284,19 +322,19 @@ declare namespace i18next {
          * options for language detection - check documentation of plugin
          * @default undefined
          */
-        detection?: object;
+        detection?: DetectionPluginOptions;
```

Comment:
I'd prefer to keep `object` until the options interfaces actually have properties. A specific name will annoy users when they goto-definition and find out that there *are* no options.

If the intent is to disable the use of these options, just make the type `{}` instead of creating a separate interface.

Reply:
This is so that i18next extensions can add / extend properties in their declaration files:

```ts
declare module 'i18next' {
    interface DetectionPluginOptions {
       a: string;
    }
}
```

`hapi` does this too for example: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/14d0a6329aaf71eaa43e6d87e43e6ec74ea17c1c/types/vision/index.d.ts#L244-L268